symforce/slam/imu_preintegration/imu_factor.h

```diff
@@ -0,0 +1,82 @@
+/* ----------------------------------------------------------------------------
+ * SymForce - Copyright 2022, Skydio, Inc.
+ * This source code is under the Apache 2.0 license found in the LICENSE file.
+ * ---------------------------------------------------------------------------- */
+
+#pragma once
+
+#include <Eigen/Dense>
+
+#include <sym/pose3.h>
+
+#include "./preintegrated_imu_measurements.h"
+
+namespace sym {
+
+/**
+ * A Functor storing on-manifold preintegrated IMU measurements between two times and
+ * which computes the local-coordinate difference between the final state (pose and
+ * velocity) and the state you would expect given the initial state and
+ * the IMU measurements.
+ *
+ * Based heavily on the on manifold ImuFactor found in GTSAM and on the paper:
+ *
+ * Christian Forster, Luca Carlone, Frank Dellaert, and Davide Scaramuzza,
+ * “IMU Preintegration on Manifold for Efficient Visual-Inertial Maximum-a-Posteriori Estimation”,
+ * Robotics: Science and Systems (RSS), 2015.
+ */
+template <typename Scalar>
+class ImuFactor {
```

Comment:
What do ya'll think of the name `ImuFactor`? This is in opposition to, say, `ManifoldImuFactor` or something like that, as a `TangentImuFactor` is likely to follow up. It's my understanding that a user is more likely to want to use this version, in which case it makes sense to simply call it `ImuFactor` to make the decision easier and more natural.

I'm not entirely sure what I think myself.